packages/react/src/components/Toolbar/ToolbarMenuItem.tsx

```diff
@@ -0,0 +1,181 @@
+import Icon from '../Icon/Icon'
+
+export interface ToolbarMenuItemProps
+  extends UIComponentProps,
+    ChildrenComponentProps,
+    ContentComponentProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   */
+  accessibility?: Accessibility
+
+  /** A toolbar item can be active. */
+  active?: boolean
+
+  /** A toolbar item can show it is currently unable to be interacted with. */
+  disabled?: boolean
+
+  /** Name or shorthand for Toolbar Item Icon */
+  icon?: ShorthandValue
+
+  /**
+   * Called on click.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onClick?: ComponentEventHandler<ToolbarMenuItemProps>
+
+  /**
+   * Called after user's focus.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onFocus?: ComponentEventHandler<ToolbarMenuItemProps>
+
+  /**
+   * Called after item blur.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onBlur?: ComponentEventHandler<ToolbarMenuItemProps>
+
+  /** Shorthand for the wrapper component. */
+  wrapper?: ShorthandValue
+}
+
+export interface ToolbarMenuItemState {
+  isFromKeyboard: boolean
+}
+
+export interface ToolbarMenuItemSlotClassNames {
+  wrapper: string
+}
+
+class ToolbarMenuItem extends UIComponent<WithAsProp<ToolbarMenuItemProps>, ToolbarMenuItemState> {
+  static displayName = 'ToolbarMenuItem'
+
+  static className = 'ui-toolbar__menuitem'
+
+  static slotClassNames: ToolbarMenuItemSlotClassNames = {
+    wrapper: `${ToolbarMenuItem.className}__wrapper`,
+  }
+
+  static create: Function
+
+  static propTypes = {
+    ...commonPropTypes.createCommon(),
+    active: PropTypes.bool,
+    disabled: PropTypes.bool,
+    icon: customPropTypes.itemShorthand,
+    onClick: PropTypes.func,
+    onFocus: PropTypes.func,
+    onBlur: PropTypes.func,
+    wrapper: PropTypes.oneOfType([PropTypes.node, PropTypes.object]),
+  }
+
+  static defaultProps = {
+    as: 'button',
+    accessibility: defaultBehavior as Accessibility,
+    wrapper: { as: 'li' },
+  }
+
+  renderComponent({ ElementType, classes, accessibility, unhandledProps }) {
+    const { children, content, disabled, icon, wrapper } = this.props
+
+    const menuItemInner = childrenExist(children) ? (
+      children
+    ) : (
+      <ElementType
+        {...accessibility.attributes.root}
+        {...unhandledProps}
+        disabled={disabled}
+        className={classes.root}
+        onBlur={this.handleBlur}
+        onFocus={this.handleFocus}
+        onClick={this.handleClick}
+      >
+        {childrenExist(children) ? (
+          children
+        ) : (
+          <>
+            {Icon.create(icon, { defaultProps: { xSpacing: !!content ? 'after' : 'none' } })}
+            {content}
+          </>
+        )}
+      </ElementType>
+    )
+
+    if (!wrapper) {
```

Comment:
```suggestion
    if (_.isNil(wrapper)) {
```

Otherwise it will fail on empty string or `0`

Reply:
This is intentional. `wrapper: false` is valid.

Reply:
`_.isNil(wrapper) || wrapper === false`?